### ISSUE TITLE
Installation failed: Failed to find a pointer-size integer type.

### ISSUE THREAD

**Issue by ptoche** [madler/zlib#494]
Specs: **macOS Catalina 10.15.4** with newest version of `xcode` (both `MacOSX10.14.sdk` and `MacOSX10.15.sdk` available in `Library/Developer/CommandLineTools/SDKs/`

Error message includes: `Installation failed: Failed to find a pointer-size integer type.`

Please let me know if more information is needed. 

        cd zlib-1.2.11
        ./configure
        Checking for gcc...
        Checking for shared library support...
        Building shared library libz.1.2.11.dylib with gcc.
        Checking for size_t... No.
        Checking for long long... Yes.
        Failed to find a pointer-size integer type.
        ** ./configure aborting.

**Comment by madler** [comment]
Odd. It works for me (macOS 10.15.4, Xcode 11.4.1). You should be getting `Checking for size_t... Yes.`.

Please attach the resulting configure.log file.

**Comment by ptoche** [comment]
Thanks for getting back to me! Below is a copy-paste:

        --------------------
        ./configure
        Sun May 17 20:55:59 PDT 2020
        Checking for gcc...
        === ztest69419.c ===
        extern int getchar();
        int hello() {return getchar();}
        ===
        gcc -c ztest69419.c
        ... using gcc

        Checking for obsessive-compulsive compiler options...
        === ztest69419.c ===
        int foo() { return 0; }
        ===
        gcc -c -O3 ztest69419.c

        Checking for shared library support...
        === ztest69419.c ===
        extern int getchar();
        int hello() {return getchar();}
        ===
        gcc -w -c -O3 -fPIC ztest69419.c
        gcc -dynamiclib -install_name ${exec_prefix}/lib/libz.1.dylib -compatibility_version 1 -current_version 1.2.11 -O3 -fPIC -o ztest69419.dylib ztest69419.o
        Building shared library libz.1.2.11.dylib with gcc.

        === ztest69419.c ===
        #include <stdio.h>
        #include <stdlib.h>
        size_t dummy = 0;
        ===
        gcc -c -O3 ztest69419.c
        ztest69419.c:1:19: fatal error: stdio.h: No such file or directory
         #include <stdio.h>
                           ^
        compilation terminated.
        (exit code 1)
        Checking for size_t... No.

        === ztest69419.c ===
        long long dummy = 0;
        ===
        gcc -c -O3 ztest69419.c
        Checking for long long... Yes.
        === ztest69419.c ===
        #include <stdio.h>
        int main(void) {
            if (sizeof(void *) <= sizeof(int)) puts("int");
            else if (sizeof(void *) <= sizeof(long)) puts("long");
            else puts("z_longlong");
            return 0;
        }
        ===
        gcc -O3 -o ztest69419 ztest69419.c
        ztest69419.c:1:19: fatal error: stdio.h: No such file or directory
         #include <stdio.h>
                           ^
        compilation terminated.
        (exit code 1)
        Failed to find a pointer-size integer type.
        ** ./configure aborting.
        --------------------

**Comment by madler** [comment]
No stdio.h. That certainly is a problem. You won't even be able to compile Hello World, let alone zlib.

Maybe you have some other gcc blocking Xcode's. Try `CC=cc ./configure`.

**Comment by ptoche** [comment]
Oh, what does that mean? 

        CC=cc ./configure
        Checking for shared library support...
        Building shared library libz.1.2.11.dylib with cc.
        Checking for size_t... No.
        Checking for long long... Yes.
        Failed to find a pointer-size integer type.
        ** ./configure aborting.

Thanks!

**Comment by ptoche** [comment]
Installation problem was resolved. I have now moved all my bash profile information to `.zshrc` and it is picked up properly by Catalina. I have access to the latest version of `gcc` and several others (gcc-10, gcc-9, and the version packaged with xcode). Right now the default gcc is `4.8.5`. I can't seem to change the default to version 10, which you would think would be the thing to do, unless I remove version 4.8.5 from the PATH altogether, which seems a bit radical. So I still have some figuring out to do. To sum up, miniconda was interfering and profiles were no longer sourced from `.bashprofile`. But the main use here is that `zlib` has been installed. This issue may therefore be closed. And thank you very much for your help.  

To install `gcc-10` I followed the steps here: https://www.youtube.com/watch?v=GUa9w5rn3a8
